content/en/docs/concepts/configuration/secret.md

```diff
@@ -672,6 +672,33 @@ A container using a Secret as a
 Secret updates.
 {{< /note >}}
 
+{{< feature-state for_k8s_version="v1.18" state="alpha" >}}
+
+For clusters that are extensively using secrets (having at least tens of thousands of unique
+Secret to Pod mounts), Kubernetes has an _Immutable Secrets/ConfigMaps_ feature. It allows you
+to mark individual Secrets/ConfigMaps as `Immutable`, which blocks any changes to their data
+and in return:
+- protects you from accidental bad updates that could cause applications outages
+- improves performance of your cluster by significantly reducing load on kube-apiserver, by
+  closing watches for secrets marked as immutable
+
+To use it, you need to enable `ImmutableEmphemeralVolumes`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) and then
+mark your Secret/ConfigMap by setting it `Immutable` field to true, e.g.:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  ...
+data:
+  ...
+immutable: true
+```
+
+Once marked as immutable, it is NOT possible to revert this change nor to mutate the contents
+of data field. You can only delete and recreate the secrete, but that will not be picked by
```

Comment:
nit:
```suggestion
of data field. You can only delete and recreate the Secret, but that will not be picked by
```

Also, I'd put this in a `{{< note >}}` or `{{< caution >}}` block to highlight the gotcha about Pod replacement.

Reply:
done